inference_schema/schema_util.py

```diff
@@ -36,6 +37,39 @@ def get_output_schema(func):
     return _get_schema_from_dictionary(OUTPUT_SCHEMA_ATTR, func)
 
 
+def get_supported_versions_for_input(func):
+    """
+    Extract the input's supported swagger versions from the decorated function.
+
+    :param func:
+    :type func: function | FunctionWrapper
+    :return:
+    :rtype: list
+    """
+    decorators = _get_decorators(func)
+    func_base_name = _get_function_full_qual_name(decorators[-1])
+    print(func_base_name)
+    print(__versions__)
+    print(__versions__.get(func_base_name, {}))
+
+    return __versions__.get(func_base_name, {}).get(INPUT_SCHEMA_ATTR, {}).get('versions', [])
+
+
+def get_supported_versions_for_output(func):
```

Comment:
I don't think there should be two functions here. Given that a function represents an http route, I think it makes more sense to say that the swagger for this route can be these versions, rather than specific input and/or output.

Reply:
combined